nabs/preprocessors.py

```diff
@@ -301,4 +302,70 @@ def daq_during_plan():
     return (yield from daq_during_plan())
 
 
+def step_size_decorator(plan):
+    """
+    Grab the last argument (number of steps), and intepret as
+    - step size if float
+    - number of steps if integer
+
+    Only works on step scans in one dimension.
+
+    Parameters
+    ----------
+    plan : plan
+        A bluesky plan that yields bluesky Msg objects.  Must be
+        a scan in one dimension, with the last argument being
+        the number of scan points / step size.
+
+    Returns
+    -------
+    step_size_plan : plan
+        The same plan as before, but modified appropriately to
+        differentiate between step size and number of steps.
+        This will be a callable generator function.
+    """
+
+    @wraps(plan)
+    def inner(*args, **kwargs):
+        if 'num' in kwargs:
+            # Currently unneeded, since daq_ascan and daq_dscan
+            # do not support num kwarg
+            n = kwargs.pop('num')
+        else:
+            # assumes (det_list, motor, start, stop, num)
+            n = args[-1]
+
+        if not isinstance(n, (int, float)):
+            raise TypeError("Step size / number of steps is "
+                            "neither float nor integer")
+
+        if type(n) is int:
+            # interpret as number of steps (default)
+            yield from plan(*args, **kwargs)
```

Comment:
Do we want to catch the return value of the plan and return it below, just in case?
```suggestion
            result = yield from plan(*args, **kwargs)
```

Reply:
What error does this catch?  I'm probably (definitely) uneducated about this but they look identical to me?

Reply:
Not an error, but the return value of the plan. This ties in with the recent discussion of getting return values from plans with `RE()` rather than a uid. Below, we would return `result` so that we don't just throw away whatever the plan is returning. Or, to be fair, we could just say `return (yield from ...)`, just syntax-wise I personally prefer the above.

The finer points of generators yielding values, sending values, and returning values is admittedly a bit hard to understand, but it's fundamental to how bluesky passes messages/return values back and forth (and does pausing and whatnot).